sale_order_price_recalculation/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# (c) 2014 Carlos Sánchez Cifuentes <csanchez@grupovermon.com>
+# (c) 2015 Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# (c) 2015 Oihane Crucelaegui <oihanecrucelaegi@avanzosc.es>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def recalculate_prices(self):
+        for line in self.mapped('order_line'):
+            order = line.order_id
+            res = line.product_id_change(
+                order.pricelist_id.id, line.product_id.id,
+                qty=line.product_uom_qty, uom=False,
+                qty_uos=line.product_uos_qty, uos=False, name=line.name,
+                partner_id=order.partner_id.id, lang=False, update_tax=True,
+                date_order=order.date_order, packaging=False,
+                fiscal_position=order.fiscal_position.id, flag=False)
```

Comment:
I'm concerned about setting uom and uos to False here, because I think product_id_change will reset these to the default product's uom and uos (not sure, and that method is complex enough for the outcome to be uncertain). 

I'd feel much reassured if you could update the automated test to have a non default unit for on the sale.order.line and check that the quantity + unit are not affected by `recalculate_prices`.